script.js

```diff
@@ -0,0 +1,746 @@
+                }
+            });
+
+            // == clear lane and buffer states and data layers
+            $.each(app.state.selRegions, function(region, selected) {
+                if (selected) {
+                    app.state.selRegions[region] = false;
+                    app.state[region].laneLayers.forEach(function (laneData) {
+                        app.activeMap.removeLayer(laneData);
+                    })
+                    app.state[region].laneData = [];
+                    app.state[region].laneLayers = [];
+                    $.each(app.state[region].selBuffers, function(buffer, state) {
+                        buffer = false;
+                    });
+                    $.each(app.state[region].bufferLayers, function(buffer, bufferData) {
+                        if (bufferData) {
+                            app.activeMap.removeLayer(bufferData);
+                            bufferData = null;
+                        }
+                    });
+                }
+            });
+
+            // == restore regions array
+            app.display.regionsArray = ["AL", "AR", "DC", "MO", "PG"];
+            console.log("  app.state: ", app.state);
+        }
+    },
+
+    // ======= laneAjaxQueue =======
+    laneAjaxQueue: function(url, region) {
+        console.log("laneAjaxQueue");
+        var laneData = null;
+
+        $.ajax({
+            url: url,
+            method: "GET",
+            dataType: "text"
+        }).done(function(jsonData){
+            console.log("*** ajax success ***");
+            var jsonData2 = $.parseJSON(jsonData);
+            var laneFeatures = L.mapbox.featureLayer(jsonData2).addTo(app.activeMap);
+            app.state[region].laneLayers.push(laneFeatures);
+            laneFeatures.setStyle(app.state[region].bikeLaneStyle);
+            if ((app.display.loopCount < app.state[region].laneData.length - 1) && (app.state[region].laneData.length != 1)) {
+                console.log("multiple files");
+                app.display.loopCount++;
+                laneData = app.state[region].laneData[app.display.loopCount];
+                url = "bikelanes/" + laneData;
+                app.laneAjaxQueue(url, region);
+            } else {
+                console.log("single file");
+                app.display.loopCount = 0;
+                if (app.display.allRegions) {
+                    if (app.display.regionsArray.length > 0) {
+                        region = app.display.regionsArray[0];
+                        laneData = app.state[region].laneData[app.display.loopCount];
+                        app.display.regionsArray.shift();
+                        if (laneData) {
+                            url = "bikelanes/" + laneData;
+                            app.laneAjaxQueue(url, region);
+                        } else {
+                            console.log("=== ALERT: Missing data for", region);
```

Comment:
We can keep this in this current PR, but eventually we should move from an error in the console (which only developers would check) to a user-friendly error in the UI (which the audience of our project would see.

We should create a GitHub issue for this after we merge this PR.